### ISSUE TITLE
Feature: An option to disabled funneling into world portals

### ISSUE THREAD

**Issue by Doclic** [StrataSource/Portal-2-Community-Edition#1345]
### Which component should be improved?

Entity

### Describe your suggestion

When `sv_player_funnel_into_portals` is set to 1, players will get funneled into the middle of `linked_portal_door`s.
This could probably maybe be useful somewhere, but, in my experience, world portals are often used to seamlessly transition between 2 places, without the player even noticing the world portal, and funneling the player into the middle of it makes it way easier to notice.
Also, a Valve dev said [here](https://invidious.kavin.rocks/watch?v=ivyseNMVt-4&t=2182) that "the idea of [portal funneling] is to reduce a bunch of the player frustration, and to reduce a bunch of false negatives that player get.", but if we just want to get seamless transitions, funneling reduces neither of these 2.

Example map to test this: (GZipped because GitHub)
[funnel_test.vmf.gz](https://github.com/StrataSource/Portal-2-Community-Edition/files/11646783/funnel_test.vmf.gz)
[funnel_test.bzp.gz](https://github.com/StrataSource/Portal-2-Community-Edition/files/11646784/funnel_test.bzp.gz)
Use noclip to go in the corner of the room.
Toggle `sv_player_funnel_into_portals` to see what it would look like without funneling.

So what I'm suggesting is adding a keyvalue to `linked_portal_door` that prevents players from getting funneled.

tl;dr: add keyvalue to `linked_portal_door` that prevents players from getting funneled

**Comment by craftablescience** [comment]
Just noting the keyvalue should probably be disabled in code and enabled in the fgd for compat (notably reloaded uses world portal funneling)

**Comment by Luke18033** [comment]
Wouldn't it be the opposite?

**Comment by craftablescience** [comment]
Oh I was thinking a keyvalue to disable funneling lol, we have it reversed

**Comment by Luke18033** [comment]
Could also check the BSP version like it does for projected textures

**Comment by craftablescience** [comment]
Yeah that's a good idea

**Comment by Luke18033** [comment]
This also affects physics objects
